docs/source/training.rst

```diff
@@ -10,290 +10,364 @@
+:obj:`"test"` split for validation.
+
+To preprocess our data, we will need a tokenizer:
 
 .. code-block:: python
 
-    from transformers import BertForSequenceClassification
-    model = BertForSequenceClassification.from_pretrained('bert-base-uncased')
-    model.train()
+    from transformers import AutoTokenizer
 
-This is useful because it allows us to make use of the pre-trained BERT encoder and easily train it on whatever
-sequence classification dataset we choose. We can use any PyTorch optimizer, but our library also provides the
-:func:`~transformers.AdamW` optimizer which implements gradient bias correction as well as weight decay.
+    tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
+
+As we saw in :doc:`preprocessing`, we can prepare the text inputs for the model with the following command:
 
 .. code-block:: python
 
-    from transformers import AdamW
-    optimizer = AdamW(model.parameters(), lr=1e-5)
+    inputs = tokenizer(sentences, padding="max_length", truncation=True)
 
-The optimizer allows us to apply different hyperpameters for specific parameter groups. For example, we can apply
-weight decay to all parameters other than bias and layer normalization terms:
+This will make all the samples have the maximum length the model can accept (here 512), either by padding or truncating
+them.
+
+However, we can instead apply these preprocessing steps to all the splits of our dataset at once by using the
+:obj:`map` method:
+
+.. code-block:: python
+
+    def tokenize_function(examples):
+        return tokenizer(examples["text"], padding="max_length", truncation=True)
+
+    tokenized_datasets = raw_datasets.map(tokenize_function, batched=True)
+
+You can learn more about the map method or the other ways to preprocess the data in the 🤗 Datasets `documentation
+<https://huggingface.co/docs/datasets/>`__.
+
+Next we will generate a small subset of the training and validation set, to enable faster training:
 
 .. code-block:: python
 
-    no_decay = ['bias', 'LayerNorm.weight']
-    optimizer_grouped_parameters = [
-        {'params': [p for n, p in model.named_parameters() if not any(nd in n for nd in no_decay)], 'weight_decay': 0.01},
-        {'params': [p for n, p in model.named_parameters() if any(nd in n for nd in no_decay)], 'weight_decay': 0.0}
-    ]
-    optimizer = AdamW(optimizer_grouped_parameters, lr=1e-5)
+    small_train_dataset = tokenized_datasets["train"].shuffle(seed=42).select(range(1000)) 
+    small_eval_dataset = tokenized_datasets["test"].shuffle(seed=42).select(range(1000)) 
+    full_train_dataset = tokenized_datasets["train"]
+    full_eval_dataset = tokenized_datasets["test"]
+
+In all the examples below, we will always use :obj:`small_train_dataset` and :obj:`small_eval_dataset`. Just replace
+them by their `full` equivalent to train or evaluate on the full dataset.
+
+.. _trainer:
+
+Fine-tuning in PyTorch with Trainer
```

Comment:
You prefer `with Trainer` to `with the Trainer`?

Reply:
We're already at two lines in the navbar, so let's even go for `with the Trainer API`